MonoGame.Framework/Graphics/SpriteBatch.cs

```diff
@@ -355,9 +355,25 @@ void CheckValid(SpriteFont spriteFont, StringBuilder text)
 		{
 			var item = _batcher.CreateBatchItem();
 
-			item.Depth = depth;
 			item.Texture = texture;
 
+            // set SortKey based on SpriteSortMode.
+            switch ( _sortMode )
+            {
+                // Comparison of Texture objects.
+                case SpriteSortMode.Texture:
+                    item.SortKey = texture.SortingKey;
+                    break;
+                // Comparison of Depth
+                case SpriteSortMode.FrontToBack:
+                    item.SortKey = depth;
+                    break;
+                // Comparison of Depth in reverse
+                case SpriteSortMode.BackToFront:
+                    item.SortKey = -depth;
+                    break;
+            }
+
 			if (sourceRectangle.HasValue) {
```

Comment:
@nkast - I just noticed another small optimization here.  We can skip the `tempRect` and a bunch of math in the case where there isn't a `sourceRectangle`:

```
if (sourceRectangle.HasValue) {
	_tempRect = sourceRectangle.Value;
	_texCoordTL.X = _tempRect.X / (float)texture.Width;
	_texCoordTL.Y = _tempRect.Y / (float)texture.Height;
	_texCoordBR.X = (_tempRect.X + _tempRect.Width) / (float)texture.Width;
	_texCoordBR.Y = (_tempRect.Y + _tempRect.Height) / (float)texture.Height;
} else {
	_texCoordTL.X = 0;
	_texCoordTL.Y = 0;
	_texCoordBR.X = 1.0f;
	_texCoordBR.Y = 1.0f;
}
```